task/backend/analytical_storage.go

```diff
@@ -178,10 +178,11 @@ func (as *AnalyticalStorage) FindRuns(ctx context.Context, filter influxdb.RunFi
 	  |> range(start: -14d)
 	  |> filter(fn: (r) => r._measurement == "runs" and r.taskID == %q)
 	  %s
+	  |> limit(n:%d)
 	  |> group(columns: ["taskID"])
 	  |> pivot(rowKey:["_time"], columnKey: ["_field"], valueColumn: "_value")
 
-	  `, filter.Task.String(), filterPart)
+	  `, filter.Task.String(), filterPart, filter.Limit-len(runs))
```

Comment:
Could this have a problem if it is a negative number?

Reply:
ohh possibly, let me check

Reply:
there's this check a few lines before the script:
```
if len(runs) >= filter.Limit {
		return runs, n, err
	}
```

so it shouldn't ever be negative